pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2555,7 +2556,7 @@ private CompletableFuture<Boolean> verifyTxnOwnership(TxnID txnID) {
                     } else {
                         return CompletableFuture.completedFuture(false);
                     }
-                });
+                }, ctx.executor());
```

Comment:
Why modify here?

Reply:
I requested this change last week because `getAuthenticationData()` was getting a non-volatile variable. However, before this PR was opened, I contributed #19507. @nicoloboschi - we can probably remove this part of the diff since reading the `authenticationData` is now thread safe.

In the long term, it might be worth exploring if we should only read this metadata from the event loop, but for now, these variables are volatile.